File: golang/api/business/business.go
package business

import (
	"api/web-service-gin/api/integrators"
	"errors"
	"fmt"
)

type BusinessLayer struct {
	wac integrators.WeatherAPIClient
}

func NewBusinessLayer(wac integrators.WeatherAPIClient) BusinessLayer {
	return BusinessLayer{wac: wac}
}

// Functions
func (bl BusinessLayer) GetForecast(
	longitude float32,
	latitude float32,
) (*GetForecastResult, error) {
	forecast, err := bl.wac.GetForecast(longitude, latitude)
	if err != nil {
		return nil, errors.New("error getting forecast")
	}

	shortForecast := forecast.ShortForecast
	temperature := fmt.Sprintf("%v %v", forecast.Temperature, forecast.TemperatureUnit)
	characterization := getCharacterization(forecast.Temperature)

	return &GetForecastResult{
		ShortForecast:    shortForecast,
		Temperature:      temperature,
		Characterization: characterization,
	}, nil
}

func getCharacterization(temperature int32) string {
	if temperature > 80 {
		return "hot"
	}
	if temperature > 70 {
		return "moderate"
	}

	return "cold"
}

File: python/api/business.py
from option import Result, Ok, Err

from integrators import WeatherAPIClient


def get_forecast(
    client: WeatherAPIClient,
    longitude: float,
    latitude: float,
) -> Result[dict, str]:
    result = client.get_forecast(longitude=longitude, latitude=latitude)
    if result.is_err:
        return Err("Error getting forecast")
    
    forecast = result.unwrap()

    short_forecast = forecast["short_forecast"]
    temperature = f"{forecast['temperature']} {forecast['temperature_unit']}"
    characterization = get_characterization(temperature=forecast["temperature"])

    return Ok(
        {
            "short_forecast": short_forecast,
            "temperature": temperature,
            "characterization": characterization,
        }
    )


def get_characterization(temperature: int) -> str:
    if temperature > 80:
        return "hot"
    if temperature > 70:
        return "moderate"
    
    return "cold"
